spec/models/item_spec.rb

```diff
@@ -6,4 +6,37 @@
     it { should have_many :invoice_items }
     it { should have_many(:invoices).through(:invoice_items) }
   end
+
+
+  describe 'instance methods' do
+    before(:each) do
+      @merchant_1 = Merchant.create!(name: 'Ron Swanson')
+
+      @item_1 = @merchant_1.items.create!(name: "Necklace", description: "A thing around your neck", unit_price: 1000)
+
+      @customer_1 = Customer.create!(first_name: "Billy", last_name: "Joel")
+      @customer_2 = Customer.create!(first_name: "Britney", last_name: "Spears")
+      @customer_3 = Customer.create!(first_name: "Prince", last_name: "Mononym")
+
+      @invoice_1 = @customer_1.invoices.create!(status: 1, created_at: '2012-03-25 09:54:09')
+      @invoice_2 = @customer_2.invoices.create!(status: 1, created_at: '2012-04-25 08:54:09')
+      @invoice_3 = @customer_3.invoices.create!(status: 1, created_at: '2012-10-25 04:54:09')
+      @invoice_5 = @customer_1.invoices.create!(status: 1, created_at: '2012-03-28 12:54:09')
+      @invoice_6 = @customer_1.invoices.create!(status: 1, created_at: '2012-03-29 12:54:09')
+      @invoice_7 = @customer_1.invoices.create!(status: 1, created_at: '2012-05-28 12:54:09')
+
+      @invoice_item_1 = InvoiceItem.create!(item_id: @item_1.id, invoice_id: @invoice_1.id, unit_price: 1000, quantity: 3, status: 0)
+      @invoice_item_2 = InvoiceItem.create!(item_id: @item_1.id, invoice_id: @invoice_2.id, unit_price: 1000, quantity: 4, status: 0)
+      @invoice_item_3 = InvoiceItem.create!(item_id: @item_1.id, invoice_id: @invoice_3.id, unit_price: 1000, quantity: 1, status: 0)
+      @invoice_item_4 = InvoiceItem.create!(item_id: @item_1.id, invoice_id: @invoice_5.id, unit_price: 1000, quantity: 9, status: 0)
+      @invoice_item_5 = InvoiceItem.create!(item_id: @item_1.id, invoice_id: @invoice_6.id, unit_price: 1000, quantity: 2, status: 0)
+      @invoice_item_6 = InvoiceItem.create!(item_id: @item_1.id, invoice_id: @invoice_7.id, unit_price: 1000, quantity: 9, status: 0)
+    end
+
+    describe '#date_with_most_sales' do
+      it 'shows date that the items sold the most and returns most recent date if multiple days have equal sales' do
+        expect(@item_1.date_with_most_sales).to eq("05/28/2012")
```

Comment:
Is there a way to test that it returns the most recent date when two days have the same quantity. Like, I think it might be good if there is a clear day that has the most sales and our first test returns that (since it is a clear winner). Then inside a second it-block we could make a bunch of sales that occurred on a different day that is more recent and expect the method to return that (as opposed to the clear winner outlined in our before(:each) data. Then we would be explicitly testing that it returns the highest one, and then explicitly checking that it returns the most recent highest one when there is a tie.

Reply:
I can see the benefit to adding an additional it statement to more explicitly test those two things, and I understand wanting to limit the test data if we can test both in one shot. We can discuss this more and decide the best way to move forward!

Reply:
Now we have two tests that make this more clear.